src/ebay-tri-state-checkbox/tri-state-checkbox.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { ChangeEvent, cloneElement, ComponentProps, FC, FocusEvent, KeyboardEvent, useState } from 'react'
+import classNames from 'classnames'
+import { EbayIcon } from '../ebay-icon'
+import { EbayLabel, EbayLabelProps } from '../ebay-field'
+import { findComponent } from '../common/component-utils'
+import { EbayChangeEventHandler, EbayFocusEventHandler, EbayKeyboardEventHandler } from '../common/event-utils/types'
+
+type Size = 'default' | 'large'
+type CheckboxState = 'true' | 'false' | 'mixed'
+type InputProps = Omit<ComponentProps<'input'>, 'size' | 'onChange' | 'onFocus' | 'onKeyDown'>
+type EbayTristateCheckboxProps = {
+    checked?: CheckboxState,
```

Comment:
We need to support the component as a controlled and as uncotrolled. Controlled means that the application maintains the state of the checkbox (by using the `checked` property) and our component will render based on the `checked` property.
Uncontrolled means that we control the state of the input, and that is defined by the `defaultChecked` property that define the initial state (argument passed on `useState()`) and we update accordingly. Marko combine both, but in react we have this control vs uncontrol [pattern](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable)